pkg/controller/replication_controller_test.go

```diff
@@ -450,3 +432,321 @@ func TestSortingActivePods(t *testing.T) {
 		}
 	}
 }
+
+// NewFakeRCExpectationsLookup creates a fake store for PodExpectations.
+func NewFakeRCExpectationsLookup(ttl time.Duration) (*RCExpectations, *util.FakeClock) {
+	fakeTime := time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC)
+	fakeClock := &util.FakeClock{fakeTime}
+	ttlPolicy := &cache.TTLPolicy{ttl, fakeClock}
+	ttlStore := cache.NewFakeExpirationStore(
+		expKeyFunc, nil, ttlPolicy, fakeClock)
+	return &RCExpectations{ttlStore}, fakeClock
+}
+
+func TestRCExpectations(t *testing.T) {
+	ttl := 30 * time.Second
+	e, fakeClock := NewFakeRCExpectationsLookup(ttl)
+	adds, dels := 10, 30
+	rc := newReplicationController(nil, 1)
+
+	// RC fires off adds and deletes at apiserver, then sets expectations
```

Comment:
Yes i meant simulating, as a way to document what happens in production. Yes currently we can't really set both to a positive value unless there's a race since we go to sleep after setting one and wake up when it's either expired or < 0. if a bunch of kubectl adds/deletes happen the counters would go negative. But I wanted to leave the expectations implementation open ended, in case I need it in other controllers. I should probably flag that case in the logs.